From 674b19fe005d675df6615ef38418f09a6af9beae Mon Sep 17 00:00:00 2001
From: Cristina Ciocan <cristina-mihaela.ciocan@nxp.com>
Date: Thu, 20 Apr 2017 13:32:43 +0300
Subject: [PATCH 1717/5242] MLK-14548: video: fb: hdmi: Allow reset pin
 deferral

commit  56fc5e8e814404a7ebda689406bd393c616d89cd from
https://source.codeaurora.org/external/imx/linux-imx.git

The HDMI transmitter Sil902x driver displays a warning when it cannot
reset the device before any other instruction, but carries on without
exiting the probe() function.

If the needed reset pin is not yet available because the pinctrl
controllers are not set up and the reset fails with deferral error, try
to acquire the pin later on in order to allow proper initialization of the
Sil902 HDMI transmitter.

Signed-off-by: Cristina Ciocan <cristina-mihaela.ciocan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mxsfb_sii902x.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/video/fbdev/mxc/mxsfb_sii902x.c b/drivers/video/fbdev/mxc/mxsfb_sii902x.c
index 97c46b6..c3c4a18 100644
--- a/drivers/video/fbdev/mxc/mxsfb_sii902x.c
+++ b/drivers/video/fbdev/mxc/mxsfb_sii902x.c
@@ -398,6 +398,8 @@ static int sii902x_probe(struct i2c_client *client,
 	ret = device_reset(&sii902x.client->dev);
 	if (ret)
 		dev_warn(&sii902x.client->dev, "No reset pin found\n");
+	if (ret == -EPROBE_DEFER)
+		return ret;
 
 	/* Set 902x in hardware TPI mode on and jump out of D3 state */
 	if (i2c_smbus_write_byte_data(sii902x.client, 0xc7, 0x00) < 0) {
-- 
1.7.9.5

